src/corona_hakab_model/manager.py

```diff
@@ -158,37 +162,111 @@ def step(self):
+            # Get number of days agent is isolated
+            if self.agents_in_isolation[agent_index] != IsolationTypes.NONE:
+                # TODO: Need to take into account when got out of isolation also
+                number_of_days_isolated = max(0,
+                                              self.current_step - self.step_to_isolate_agent[agent_index])
+            else:
+                number_of_days_isolated = 0
+            weekly_connection_isolation_ratio = 1 - min(number_of_days_isolated / 7, 1)
+            for connected_agents in self.connection_data.connected_ids_by_strength[agent_index].values():
+                for daily_agent_id in connected_agents.daily_connections:  # All of this need to be home isolated
+                    # If were not isolated, and not sick, needs to
+                    if self.agents_in_isolation[daily_agent_id] == IsolationTypes.NONE and \
+                            not self.tested_positive_vector[daily_agent_id]:
+                        if self.step_to_isolate_agent[daily_agent_id] < self.current_step:
+                            self.step_to_isolate_agent[daily_agent_id] = self.current_step + \
+                                                                         self.consts.isolate_after_num_day
+                agents_to_iterate = list(filter(lambda index: (self.agents_in_isolation[index] == IsolationTypes.NONE)
+                                                              and not self.tested_positive_vector[index],
+                                                connected_agents.weekly_connections))
+                non_sick_agents = self.medical_machine.default_state_upon_infection.agents
+                shuffle(agents_to_iterate)
+                agents_to_iterate.sort(key=lambda index: self.agents[index] in non_sick_agents, reverse=True)
+                how_many_to_isolate = round(len(agents_to_iterate) * weekly_connection_isolation_ratio)
+                j = 0
+                for i in range(how_many_to_isolate):
+                    # If already about to get isolated, do not update it
+                    if self.step_to_isolate_agent[agents_to_iterate[i]] > self.current_step:
+                        continue
+                    self.step_to_isolate_agent[agents_to_iterate[i]] = self.current_step + \
+                                                                       self.consts.isolate_after_num_day
+                    j += 1
+            # Isolate the agent
+            self.step_to_isolate_agent[agent_index] = self.current_step + self.consts.isolate_after_num_day
+
+        # Isolating symptomatic agents
+        for agent in self.new_agents_with_symptoms:
+            # If is not getting ready to be isolated, or isolated already, then isolate
+            if self.step_to_isolate_agent[agent_index] < self.current_step:
+                self.step_to_isolate_agent[agent.index] = self.current_step + self.consts.isolate_after_num_day
```

Comment:
I think that symptomatic agents should be isolated immediately, don't you think?

Reply:
They point is that they're isolating themselves cause they feel like crap, not because someone called them and told them that they've been exposed.

Reply:
No, I do not agree because of 2 things:
*) Thats the how it was characterized
*) Think about the start of the symptoms, some people want to work, they might say things like "Yeah, I feel a lil bad, but not REALLY bad". Only after number of days of having symptoms they will say: I wanna go home. This might change with isolation policy.

Reply:
Well, it's kinda how it was characterized to me in the phone, but I agree that it's not what they wrote in the paper, so OK